src/options.rs

```diff
@@ -101,7 +108,158 @@ pub struct SessionOptions {
     ///
     /// Defaults to `true`.
-    pub publish: bool,
+    pub publish_lease_set: bool,
+
+    /// Minimum number of ElGamal/AES Session Tags before we send more. Recommended: approximately
+    /// tagsToSend * 2/3
+    /// 
+    /// /// Defaults to `30`.
+    pub crypto_low_tag_threshold: usize,
+
+    /// Inbound tag window for ECIES-X25519-AEAD-Ratchet. Local inbound tagset size.
+    /// 
+    /// /// Defaults to `160`.
+    pub crypto_ratchet_inbound_tags: usize,
+
+    /// Outbound tag window for ECIES-X25519-AEAD-Ratchet. Advisory to send to the far-end in the
+    /// options block.
+    /// 
+    /// /// Defaults to `160`.
+    pub crypto_ratchet_outbound_tags: usize,
+
+    /// Number of ElGamal/AES Session Tags to send at a time.
+    /// 
+    /// /// Defaults to `40`.
+    pub crypto_tags_to_send: usize,
+
+    /// For authorization, if required by the router.
+    pub username: Option<String>,
+
+    /// For authorization, if required by the router.
+    pub password: Option<String>,
+
+    /// If incoming zero hop tunnel is allowed
+    ///
+    /// Defaults to 'false'
+    pub inbound_allow_zero_hop: bool,
+
+    /// Number of redundant fail-over for tunnels in
+    ///
+    /// Defaults to `0`.
+    pub inbound_backup_qty: usize,
+
+    /// Number of IP bytes to match to determine if two routers should not be in the same tunnel. 0
+    /// to disable.
+    ///
+    /// Defaults to `0`.
+    pub inbound_ip_restriction: usize,
+
+    /// Random amount to add or subtract to the length of tunnels in.
+    ///
+    /// Defaults to `0`.
+    pub inbound_len_variance: isize,
+
+    /// Used for consistent peer ordering across restarts.
+    pub inbound_random_key: Option<String>,
+
+    /// If outgoing zero hop tunnel is allowed
+    ///
+    ///  Defaults to 'false'
+    pub outbound_allow_zero_hop: bool,
+
+    /// Number of redundant fail-over for tunnels out
+    ///
+    /// Defaults to `0`.
+    pub outbound_backup_qty: usize,
+
+    /// Number of IP bytes to match to determine if two routers should not be in the same tunnel. 0
+    /// to disable.
+    ///
+    /// Defaults to `0`.
+    pub outbound_ip_restriction: usize,
+
+    /// Priority adjustment for outbound messages. Higher is higher priority.
+    ///
+    /// Defaults to `0`.
+    pub outbound_priority: isize,
+
+    /// Random amount to add or subtract to the length of tunnels in.
+    ///
+    /// Defaults to `0`.
+    pub outbound_len_variance: isize,
+
+    /// Used for consistent peer ordering across restarts.
+    pub outbound_random_key: Option<String>,
+
+    /// Set to false to disable ever bundling a reply LeaseSet.
+    ///
+    /// Defaults to `true`.
+    pub should_bundle_reply_info: bool,
+
+    /// Close I2P session when idle
+    ///
+    /// Defaults to 'false'
+    pub close_on_idle: bool,
+
+    /// (ms) Idle time required
+    ///
+    /// Defaults to '1800000' ms (i.e. 30 minutes)
+    pub close_idle_time: usize,
```

Comment:
```suggestion
    /// Idle time required
    ///
    /// Defaults to 30 minutes.
    pub close_idle_time: Duration,
```

Reply:
Is the comment correct? It's the same for both `close_on_idle` and `reduce_idle_time`